tracing-subscriber/Cargo.toml

```diff
@@ -47,7 +47,9 @@ tracing-core = { path = "../tracing-core", version = "0.2", default-features = f
 
 # only required by the `env-filter` feature
 tracing = { optional = true, path = "../tracing", version = "0.2", default-features = false }
-matchers = { optional = true, version = "0.1.0" }
+matchers = { optional = true, version = "0.2.0" }
+regex = { optional = true, version = "1.9.0", default-features = false, features = ["std", "unicode-case", "unicode-perl"] }
+regex-automata = { optional = true, version = "0.4", default-features = false, features = ["std"] }
```

Comment:
I don't understand, what are the new dependencies for? They don't seem to be used.

Reply:
(`regex` is not needed, but I forgot about the details when updating.... Just pushed a new version.)

> It looks like the matchers::BuildError type doesn't implement std::error::Error, which breaks some code that uses ? on it in a method that returns Box<dyn Error + Send + Sync>: https://github.com/tokio-rs/tracing/actions/runs/9930533323/job/27719987902?pr=3033#step:7:218

Basically, regex-automata is a dependency of matchers, but not with the std feature. Hence, code that relies on BuildError implementing std::error::Error can break from this update. Now, we guarantee that the std feature is included (matchers is not no_std anyway) and things work as before.